src/main/java/io/cdap/plugin/format/DBTableRecordReader.java

```diff
@@ -157,4 +157,15 @@ public void close() throws IOException {
       throw new IOException(exception);
     }
   }
+
+  private String getQuery() {
+    if (split.getLowerClause() != null && split.getUpperClause() != null) {
+      // Now append the conditions associated with our split.
+      // Build the WHERE clauses associated with the data split.
+      String conditionClauses = "WHERE (( " + split.getLowerClause() + (" ) AND ( ") + split.getUpperClause() + " ))";
```

Comment:
This type of logic seems like it should be encapsulated in the split class and exposed in a getWhereClause() method.

Reply:
done